cattle/plugins/host_info/os_c.py

```diff
@@ -0,0 +1,44 @@
+import platform
+import cattle.utils
+
+
+class OSCollector(object):
+    def key_name(self):
+        return "osInfo"
+
+    def _zip_fields_values(self, keys, values):
+        data = {}
+        for key, value in zip(keys, values):
+            if len(value) > 0:
+                data[key] = value
+        else:
+            data[key] = None
+
+        return data
+
+    def _get_docker_version(self):
+        data = {}
+
+        if platform.system() == 'Linux':
+            data['dockerVersion'] = \
+                cattle.utils.check_output(['docker', '-v']).rstrip()
+
+        return data
+
+    def _get_os(self):
+        data = {}
+        if platform.system() == 'Linux':
+            info = platform.linux_distribution()
+            keys = ["distribution", "version", "versionDescription"]
+            data = self._zip_fields_values(keys, info)
+
+            data['kernelVersion'] = \
+                (lambda x: x if len(x) > 0 else None)(platform.release())
```

Comment:
"Technicality no down boo over"

No need to use a lambda here. I'd rather see this logic take a couple lines if need be.